### DISCUSSION TITLE
How many cores does (an older) RabbitMQ use

### DISCUSSION THREAD

**Discussion by gitzko** [rabbitmq/rabbitmq-server#8510] [Other] [answered]
Hi,

we have an older RabbitMQ 3.3.1 connected application, with Erlang 17. I see maximum CPU usage clamping at about 25%, which suggests me that it's using one of 4 vcpus to the max. 
Does anyone know whether this version of Rabbit uses multiple cores, can it be coerced into using them?

**Comment by lukebakken** [comment]
Hello, thanks for using RabbitMQ. Obviously, the version of RabbitMQ and Erlang you're using are _extremely_ out-of-date.

How many queues are you using?

**Reply by gitzko** [reply]
Around 10 durable ones, and as much auto delete.

**Comment by michaelklishin** [comment, marked as answer]
Just like 3.12, RabbitMQ 3.3 will use [a single core for every queue replica](https://rabbitmq.com/queues.html#runtime-characteristics). So if you see a node only use one core, then either your workload is dominated by a single very busy queue, or the process is limited (e.g. using cgroups) to use just one core, or the runtime hasn't detected the CPU topology correctly (I highly doubt it).
